Make the quote characters customizable.

adawolfa opened the pull request:
Some languages, such as Czech or French use quote characters different from those currently hardcoded. This commit introduces following configuration options:

- `smartpunct/double_quote_opener`
- `smartpunct/double_quote_closer`
- `smartpunct/single_quote_opener`
- `smartpunct/single_quote_closer`

Currently hardcoded character is used as a default value for each option and this is also covered by a test.

Configuration options are passed to `QuoteProcessor::createDoubleQuoteProcessor()` and `QuoteProcessor::createSingleQuoteProcessor()` during extension registration.

_EDIT: BC is now preserved._

colinodell commented on src/QuoteProcessor.php:
```diff
@@ -87,20 +87,26 @@ public function process(AbstractStringContainer $opener, AbstractStringContainer
      * @return QuoteProcessor
      */
-    public static function createDoubleQuoteProcessor(): self
+    public static function createDoubleQuoteProcessor(string $opener, string $closer): self
```
Can we add default values for these two parameters?  For example:

```
public static function createDoubleQuoteProcessor(string $opener = Quote::DOUBLE_QUOTE_OPENER, string $closer = Quote::DOUBLE_QUOTE_CLOSER): self
```

If you can do this for both static methods then we can introduce this as a backward-compatible change :)

colinodell commented on README.md:
```diff
@@ -44,6 +44,14 @@ $environment = Environment::createCommonMarkEnvironment();
+        'double_quote_opener' => '„',
+        'double_quote_closer' => '“',
+    ],
+];
```
Can you please make this show all four options and use the (current) default characters?

colinodell reviewed (changes requested):
This is a great idea - nicely done!  I only have two small requests - if you could please make those changes I'll get this merged and released!

adawolfa committed:
Make the quote characters customizable.

Some languages, such as Czech or French use quote characters different from those currently hardcoded. This commit introduces following configuration options:

- `smartpunct/double_quote_opener`
- `smartpunct/double_quote_closer`
- `smartpunct/single_quote_opener`
- `smartpunct/single_quote_closer`

Currently hardcoded character is used as a default value for each option and this is also covered by a test.

Configuration options are passed to `QuoteProcessor::createDoubleQuoteProcessor()` and `QuoteProcessor::createSingleQuoteProcessor()` during extension registration.

colinodell reviewed (approved):
This looks perfect!  Thank you so much for your contribution!

colinodell commented:
[v1.1.0](https://github.com/thephpleague/commonmark-ext-smartpunct/releases/tag/v1.1.0) has been released with this new feature.  Thanks again!
